examples/providers/examples/mock.rs

```diff
@@ -0,0 +1,82 @@
+use ethers::prelude::*;
+
+type BoxErr = Box<dyn std::error::Error>;
```

Comment:
let's use `eyre::Error` everywhere and `eyre::Result`

Reply:
Ok let's keep `eyre::Result` (I wanted to introduce some variations)